ENH: Update Pre-Processing UI to improve BVR workflow

sbelsk opened the pull request:
Resolves https://github.com/BrownBiomechanics/SlicerAutoscoperM/issues/131, resolves #145.

These changes are aimed to improve the user workflow in the Pre-Processing module, specifically targeting the improvement of the Autoscoper configuration file generation. Further changes to improve error handling to make it more user-friendly are to be continued in a separate upcoming PR.

Changes in this PR include:
* 'Volume Node' and 'Output Directory' at top of UI were swapped
* 'Trial Name' moved to the config generation section as the '<blank>'.cfg field
* The segmentation generation section was redesigned to better convey the two separate generation methods
* The 'Adavnced Options' sections was renamed to 'Default Subdirectories', moved below the 'Segmentation Generation' section, and is now rendered as collapsed by default
* The 'VRG Resolution' fields from 'Advanced Options' was moved to the config generation section
* In the 'Generate Config' section, added file and path selectors to specify the camera calibration files, radiograph root directories, and volume files to be written to the generated config file. Also added 'Voxel Size' fields that are automatically populated based on the selected input volume node.
* Elements in the .ui files were edited to be properly indexed and named

Remaining TODOs:
- [x] Decide and implement correct pairing of radiograph subfolders with their camera calibration files
- [x] Update [the documentation page](https://autoscoper.readthedocs.io/en/latest/tutorials/pre-processing-module.html) --> see https://github.com/BrownBiomechanics/Autoscoper/pull/308

| Existing Pre-Processing UI | New Pre-Processing UI |
| -------------------------- | --------------------- |
| ![image](https://github.com/user-attachments/assets/2a5b8686-7df8-408e-a91b-80d213e2d0e3) | ![image](https://github.com/user-attachments/assets/f1a5751f-a9ce-4e57-978f-53cdc906e1d0) |

amymmorton reviewed (changes requested):
-voxel size should not be manually editable- but inherited from the volume node used to gen PV
* as part of PV gen - assign the voxel size values
*create a subdirectory in Models folder 'AUT'   so that the Models level h used in Segmentation generation remains unchanged


* remove 'virtual' radiograph subdirectory

amymmorton commented:
![preproc_cfgeduts](https://github.com/user-attachments/assets/cfc50138-e650-4279-9d07-b012bd17a54d)

sbelsk commented:
@amymmorton  In the last commit pushed to this branch, I moved a few of the UI widgets so that the pre-processing tab isn't so wide, made sure that the segmentation generation options are enabled/disabled based on which radio button is selected (automatic segmentation vs. seg from models), and also implemented the file selectors for the config generation group. 

Current state of the UI:
![image](https://github.com/user-attachments/assets/a34c3461-fd6f-4e64-b517-fbed980aebb7)

amymmorton commented:
- suggest adding color contrast to push buttons throughout
- toggle on the STL Models directory label when 'Segmentation from Model' radio button active
- Trial Names -> search for subfolders in RadiographImages subdirectory
- rename 
![image](https://github.com/user-attachments/assets/ecc62879-e5d4-48e7-bc9d-b15a02e3239a)

'or enter:' label to 'AND enter:'

- @jdholt04 ?? can assume order in radiograph subfolders is same order as CamerCalibration.txt

sbelsk commented:
Remaining TODOs at this point:

- [x] toggle on the STL Models directory label when 'Segmentation from Model' radio button active
- [x] for Trial Names, search for subfolders instead of files in RadiographImages subdirectory
- [x] rename 'or enter:' label to 'AND enter:' at the top of the "Generate Config" section
- [ ] add color contrast to push buttons throughout?
- [ ] decide and implement correct pairing of radiograph subfolders with their camera calibration files
- [ ] update functionality of "Generate Config File" button using data from finalized UI design

sbelsk commented:
Current status:
* updated the list of remaining TODOs (see previous comment)
* The "Generate Config File" button now works using the data from the different UI fields (with the exception of the voxel size fields what we currently grab from the input volume node, and also with the order of the the trial names and camera calibrations being assumed to match). It should successfully write the output file and then automatically populate the "Config File" field in the "Autoscoper Control" tab as before.
* render resolution values are no longer being divided by two before writing to the generated config file
* minor UI tweaks made since last screenshot: 
![image](https://github.com/user-attachments/assets/c292564f-b2bd-4fb6-ae9e-4aa5513df381)

amymmorton commented:
"populate the "Voxel Size" according to the input volume node selected (should we let users have the ability to manually override it too?)"

I say No. The only options are voxel size form volume node / volume.tfm (which must both be the same)

amymmorton commented on AutoscoperM/AutoscoperM.py:
```diff
@@ -470,7 +486,7 @@ def onGenerateConfig(self):
             mainOutputDir = self.ui.mainOutputSelector.currentPath
             configFileName = self.ui.configFileName.text
 
-            configPath = os.path.join(mainOutputDir, configFileName, ".cfg")
```
neat - didn't know you could concatenate in python like that 👍

amymmorton reviewed (approved):
![image](https://github.com/user-attachments/assets/6fe9c932-4507-4680-a514-dc091e1b2e67)

![image](https://github.com/user-attachments/assets/ae343843-5793-431c-baf4-11303c038374)

amymmorton reviewed (commented):
👍

sbelsk commented:
Cleaned up the commit history on this branch for ease of reviewing

sbelsk commented:
Updated the config paths selection based on @amymmorton's draft. I ended up implementing a design that supports any number of corresponding camera file & radiograph subdir pairs (see updated screenshot in the PR summary). Feel free to review the update, particularly any UI text/labels and tooltips on the "add" buttons to see if we can better clarify their purpose.

amymmorton reviewed (changes requested):
![image](https://github.com/user-attachments/assets/d107179a-a38c-490f-8460-0f537055e314)

@jdholt04  any test updates from you from last week?...

jdholt04 commented:
@amymmorton 

I think the ui looks great. When testing I am unable to select files/subdirectories after populating the selection boxes. This could be an error on my part...

I replaced AutoscoperM.py, IO.py and AutoscoperM.ui with the changed files. Are there any additional files that I need to replace?


![Capture](https://github.com/user-attachments/assets/d3b64b19-bdd2-472e-b65b-3aff59a72114)

jdholt04 reviewed (changes requested):
I was able to successfully generate a config file using the changes made to the pre-processing module (tested in Slicer 5.6.2). Config file was generated and was able to be loaded into Autoscoper.

![image](https://github.com/user-attachments/assets/baa5d40e-5716-49e2-b5f0-b941282d4460)

![image](https://github.com/user-attachments/assets/3f6673d3-178d-42b5-96ab-5b08da127377)

One change to the UI that would be helpful: In dark mode the boxes to select files to be added to the config file are not visible. Could changes be made to the UI to make it more intuitive for users in dark mode? 

![image](https://github.com/user-attachments/assets/c39a629e-ea55-4b4e-9b91-35a039921476)
![image](https://github.com/user-attachments/assets/80044b33-e186-4236-a57c-1247b7bc1f9d)

sbelsk commented:
Thanks for the feedback! I updated the checkbox background color so it stands out more in dark mode, following [this fix](https://github.com/Slicer/Slicer/pull/7098) in a [similar issue in Slicer](https://github.com/Slicer/Slicer/issues/5185).

![image](https://github.com/user-attachments/assets/b2e18b42-30fd-471f-8423-cc9d4f2fad4a)
![image](https://github.com/user-attachments/assets/377fec1c-5e4c-490f-9aaf-da25a600d23e)

sbelsk commented:
Force pushed to make the commit checks pass
